napari/_qt/qt_main_window.py

```diff
@@ -1729,6 +1730,65 @@ def export_figure(
             imsave(path, img)
         return img
 
+    def export_rois(
+        self,
+        rois: list[np.ndarray],
+        paths: Optional[list[str]] = None,
+        scale: Optional[float] = None,
+    ):
+        """Export the shapes rois with storage file paths.
+
+        For each shape, moves the camera to the center of the shape
+        and adjust the canvas size to fit the shape.
+        Note: The shape height and width can be of type float.
+        However, the canvas size only accepts a tuple of integers.
+        This can result in slight misalignment.
+
+        Parameters
+        ----------
+        rois: list[np.ndarray]
+            A list of arrays  with each being of shape (4, 2) representing a rectangular roi.
+        paths: list
+            The list to store file path for shapes roi
+
+        Returns
+        -------
+        screenshot_list: list
+            The list with roi screenshots.
+
+        """
+        if paths is not None and len(paths) != len(rois):
```

Comment:
Hmm I think it would be nice if `paths` is `None` to just make paths for the user and save to local working directory in a folder maybe? It was annoying having to create the paths myself. Or maybe we should allow passing a string in which case we name the individual pngs ourselves and save to the given path.

Reply:
Thanks for the feedback and I agree that would make for a much better UX. We can save in current working directory using a `folder_name=""`string passed in the function when `path` is `none`. Naming would look something like `folder_name/f"roi_{idx+1}.png"`